### DISCUSSION TITLE
How to setup Mobile/Desktop separate bundles with nextjs

### DISCUSSION THREAD

**Discussion by mohitgahlotigp** [vercel/next.js#29391] [Help]
I am working on a new project, it has both desktop and mobile site. The project is Ecom based. 

As i have opted nextjs, looking for best approach to split my desktop/mobile site bundles. 

Thinking to use monorepo based architecture along with below packages 
- redux 
- redux saga
- Axios or SWR Or Fetch 

Looking forward for best approaches so can have a good setup for our new project. 

Also would be great if can inline first fold css for each router.

**Comment by alex-misch** [comment]
Hi! I found the solution for separating code for mobile/desktop in NextJS 12:
1. Create route with two versions like `/auth/desktop.tsx` and `/auth/mobile.tsx`. 
2. Configure your next.config.js like below
```
const rewriteToMobile = (...has) => ({
  source: '/:path*',
  destination: '/:path*/mobile',
  has,
});

const rewriteToDesktop = (...has) => ({
  source: '/:path*',
  destination: '/:path*/desktop',
  has,
});

module.exports = {
  rewrites() {
    return [
      rewriteToMobile({type: 'query', key: 'siteversion', value: 'mobile'}),
      rewriteToDesktop({type: 'query', key: 'siteversion', value: 'desktop'}),
      rewriteToMobile({type: 'cookie', key: 'siteversion', value: 'mobile'}),
      rewriteToDesktop({type: 'cookie', key: 'siteversion', value: 'desktop'}),
    ];
  },
};

```
3. Add _middleware.ts to your pages folder and write logic that sets initial siteversion cookie
